### ISSUE TITLE
$_SESSION [siteID] not cleared/set with multiple installs

### ISSUE THREAD

**Issue by siramsay** [chishiki/perihelion#24]
When I work on one Perihelion site instance and then move to another site instance the first instance $_SESSION array is used by the second site. 

This means that data seems "missing" as the siteID in the content table doesn't match and therefore the SQL query returns nothing. 

Edgecase:
This may happen with the usage of Perihelion when company divisions are split between two installs. 

![session_data_not_cleared](https://user-images.githubusercontent.com/16264981/126252632-e0c7e648-90d3-45c2-8052-03fa0896a973.png)

**Comment by chishiki** [comment]
@siramsay I notice that the site on the left is localhost... I wonder if this is related to #1

**Comment by siramsay** [comment]
> @siramsay I notice that the site on the left is localhost... I wonder if this is related to #1

Somewhat different as I can log out and then log back in and a new session with the correct ID is set. 

I also don't think it matters if the sites are on the same host, the other day I kind of had the same issue and it was using the ID from one site and posting to the database of the other and then the data wasn't rendering. 
<img width="963" alt="Screen Shot 2021-07-13 at 7 06 22 PM" src="https://user-images.githubusercontent.com/16264981/126273463-53faa21a-1436-4d3c-a804-df366141f4d9.png">

I haven't looked at this closely to see what is happening exactly but thought you may have some idea for a fix.

**Comment by siramsay** [comment]
I guess testing 2 sites on remote servers would answer your question definitively.

**Comment by chishiki** [comment]
Hmmm... yeah I've not seen a case of sessions being shared between sites before. If you have multiple instances with different configurations using the same database though I'm not sure what would happen. We typically don't have multiple instances using the same database so maybe you've stumbled onto something.

Double check your vault/config/instance.config.php files for sure, since I know you don't have any multi-site deployments.

I suspect that the code referenced in #1 works as expected on the server (using the URL to resolve siteID to 104 in perihelion_Site.siteID) but on localhost the code doesn't know what site it's on so it defaults to 1 or something. If they both share the same DB then this could manifest as two different siteIDs (1 & 104) using the same DB.

Just a guess though. We can work through it in our next meeting?

**Comment by siramsay** [comment]
yes, I guess it is related to #1, sorry I was thinking of another localhost issue and hadn't seen that one.

It would seem that on bootstrap the config isn't being read, after thinking about it, so same conclusion. If it was read and then checked against the $SESSION stored, if it didn't match you should be logged out with a message and directed to the log in page. 

Close this issue or combine I guess.
